src/scenes/gamescreen.ts

```diff
@@ -61,4 +86,41 @@ export class GameScreen extends Phaser.Scene {
         }
         this.dragStart = activePointer.position.clone();
     }
+
+    /**
+     * All tiles are kept in a placedTilesContainer.\
+     * To update display, first remove all existing images, then create new ones using placed tiles from Tilemanager\
+     *
+     * Whenever information about tiles is updated, it calls a list of placed tiles from TileManager
+     *
+     * Add the image at coordinate * pixels
+     *
+     * @returns void
+     */
+    private displayPlacedTiles(): void {
+        assert(this.placedTilesContainer);
+        this.placedTilesContainer.removeAll(true);
+
+        const placedTiles: Nullable<Tile[]> = TileManager.getPlaced();
+        if (!placedTiles) {
+            return;
+        }
```

Comment:
I think this should never be the case, you can remove that if and only have the assert below.